src/cromshell/cost/command.py

```diff
@@ -0,0 +1,302 @@
+    # the table to datafram or CSV. Instead, this loop will get the first row and break.
+
+    if detailed:
+        for row in query_results:
+            print("{}\t{}\t{}\t{}".format(row.value, row.description, row.task_name,
+                                              row.cost))
+
+    else:
+        for row in query_results:
+            total_cost = round(float(row.cost), 2)
+            print(f"${total_cost}")
+            print("Costs rounded to nearest cent (approximately).")
+
+
+def query_bigquery(
+    workflow_id: str,
+    bq_database: str,
+    start_date: str,
+    end_date: str,
+    detailed: bool
+):
+    """
+        Query bigqueury for cost
+
+    :param workflow_id:
+    :param bq_database:
+    :param start_date:
+    :param end_date:
+    :return:
+    """
+    from google.cloud import bigquery
+    client = bigquery.Client()
+
+    # Todo: remove LIMIT in query
+    if detailed:
+        query_job = client.query(
+            f"""
+                SELECT wfid.value, service.description, task.value as task_name, sum(cost) as cost
+                FROM `{bq_database}` as billing, UNNEST(labels) as wfid, UNNEST(labels) as task
+                WHERE cost > 0
+                AND task.key LIKE "wdl-task-name"
+                AND wfid.key LIKE "cromwell-workflow-id"
+                AND wfid.value like "%{workflow_id}"
+                AND partition_time BETWEEN "{start_date}" AND "{end_date}"
+                GROUP BY 1,2,3
+                ORDER BY 4 DESC
+                LIMIT 100
+            """
+        )
+    else:
+        query_job = client.query(
+            f"""
+                SELECT sum(cost) as cost
+                FROM `{bq_database}`, UNNEST(labels)
+                WHERE value = "cromwell-{workflow_id}" AND partition_time BETWEEN "{start_date}" AND "{end_date}"
+                LIMIT 100
+            """
+        )
+
+    query_results = query_job.result()
+
+    if query_results.total_rows == 0:
+        LOGGER.error("Could not retrieve cost - no cost entries found.")
+        raise ValueError("Could not retrieve cost - no cost entries found.")
+    else:
+        return query_results
+
+
+def minimum_time_passed_since_workflow_completion(
+    end_time: str, min_hours: int = 24
+) -> (bool, datetime):
+    """
+    Make sure 24 hours have passed between job finish time and executing this command
+    (finished time minus current time).
+
+    :param end_time: workflow completion time obtained from cromwell job metadata
+    :param min_hours:
+    :return:
+    """
+
+    delta = datetime.now(timezone.utc) - datetime.strptime(
+        end_time, "%Y-%m-%dT%H:%M:%S.%f%z"
+    )
+
+    mindelta = timedelta(hours=min_hours)
+
+    return (True, delta) if delta > mindelta else (False, delta)
+
+
+def check_cost_table_is_configured(config_options: dict) -> None:
+    if "bq_cost_table" not in config_options:
+        raise KeyError('Cromshell config file is missing "bq_cost_table" Key')
```

Comment:
This should probably be a custom error of some sort?  It's something correctable by the user. (maybe print the path to the config file that's in use as well?)